### PR TITLE
test: Moves disk_size_gb to replication spec for GetClusterInfo

### PR DESCRIPTION
**Pull request by EspenAlbert**
## Description

Moves disk_size_gb to replication spec for GetClusterInfo

Link to any related issue(s): CLOUDP-261482

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR. A migration guide must be created or updated if the new feature will go in a major version.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR. A migration guide must be created or updated.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contributing guides](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/contributing/README.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

### CODE REVIEW

**Review by lantoli — APPROVED** [review]
good!  do we need to do anything with num_shards or not needed for ClusterInfo?

**Comment by EspenAlbert** [comment]
@lantoli no, `num_shards` are not used directly in any of the examples anymore 😁

tl;dr: All failing tests are ok
### SUMMARY OF FAILING TESTS
- TestAccNetworkRSNetworkPeering_Azure has 1 failures: **fixed by [deleting existing peering connection manually](https://mongodb.slack.com/archives/C06Q7QF1ZST/p1721658167936409?thread_ts=1721610202.657029&cid=C06Q7QF1ZST) )**
  - [a day ago failed in 5 minutes](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/10035932088/job/27732901767#step:5:227)
- TestAccPrivateEndpointRegionalMode_conn has 1 failures: **flaky test, disabled in master**
  - [a day ago failed in 19 minutes](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/10035932088/job/27732901767#step:5:303)
